### PR TITLE
Add integration tests for Terraform

### PR DESCRIPTION
**Pull request by cmilam87**
Adding in integration tests for Terraform to test backend changes:

- Change backend from local to S3
- Change AWS provider version from 1 to 2

There is another test commented out that will test changing the backend from local (none specified) to local (path specified). This however is always failing in Terraform and creates the resources each time regardless of the path or `-reconfigure` option.

### CODE REVIEW

File: tests/integration/terraform/terraform.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        test_name += '.test'
+
+    if os.path.isfile(os.path.join(TERRAFORM_TESTS_DIR, test_name)):
+        LOGGER.debug('Copying Terraform test: "%s" to main.tf', test_name)
+        shutil.copy(os.path.join(TERRAFORM_TESTS_DIR, test_name),
+                    os.path.join(TERRAFORM_TF_DIR, 'main.tf'))
+
+        LOGGER.info('Running test "%s"', test_name)
+        with change_dir(WORKING_DIR):
+            subprocess.check_call('runway deploy', shell=True)
```
Comment (troyready) [inline tests/integration/terraform/terraform.py]: Does this work instead:
```
subprocess.check_call(['runway', 'deploy'])
```
?
(to get rid of the `shell=True`)

File: tests/integration/terraform/terraform.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+def teardown():
+    """Teardown all resources."""
+    # terraform destory
+    # stacker destory
+    clean()
+    run_stacker('destroy')
+
+
+def run_tests():
```
Comment (troyready) [inline tests/integration/terraform/terraform.py]: Is any of this appropriate to surround in a `try/finally` block? I'm wondering how much we can do to prevent a bad merge from getting the test environment in a state that requires manual intervention.

File: tests/integration/terraform/test_terraform.tf/.terraform-version [hunk 3]
```diff
@@ -0,0 +1 @@
+0.11.13
```
Comment (troyready) [inline tests/integration/terraform/test_terraform.tf/.terraform-version]: Can we extend this to test multiple Terraform versions (e.g. so we can test both 0.11.13 and 0.12.6)?

File: tests/integration/terraform/tf_state.cfn/templates/tf_state.yaml [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+  BucketName:
+    Default: ''
+    Description: (optional) Name for the S3 bucket
+    Type: String
+  TableName:
+    Default: ''
+    Description: (optional) Name for the DynamoDB table
+    Type: String
+Resources:
+  ManagementPolicy:
```
Comment (troyready) [inline tests/integration/terraform/tf_state.cfn/templates/tf_state.yaml]: Let's delete the `ManagementPolicy`  from the template; it's not used in this context and will make later permission scoping easier

File: tests/integration/terraform/terraform.py (hunk 1 above)
Comment (cmilam87) [reply tests/integration/terraform/terraform.py]: Yep - this works.

File: tests/integration/terraform/terraform.py (hunk 2 above)
Comment (cmilam87) [reply tests/integration/terraform/terraform.py]: Wrapped in try/except/finally

File: tests/integration/terraform/test_terraform.tf/.terraform-version (hunk 3 above)
Comment (cmilam87) [reply tests/integration/terraform/test_terraform.tf/.terraform-version]: added test for terraform version

File: tests/integration/terraform/tf_state.cfn/templates/tf_state.yaml (hunk 4 above)
Comment (cmilam87) [reply tests/integration/terraform/tf_state.cfn/templates/tf_state.yaml]: removed

**Comment by cmilam87** [comment]
Completely refactored all the integration tests into a better testing suite. More tests can be added later. Right now it's only testing the provider version change and the terraform version change.

**Comment by cmilam87** [comment]
Output from CodeBuild:
```[Container] 2019/08/13 21:12:17 Waiting for agent ping 
[Container] 2019/08/13 21:12:20 Waiting for DOWNLOAD_SOURCE 
[Container] 2019/08/13 21:12:21 Phase is DOWNLOAD_SOURCE 
[Container] 2019/08/13 21:12:21 CODEBUILD_SRC_DIR=/codebuild/output/src960133886/src/github.com/cmilam87/runway 
[Container] 2019/08/13 21:12:21 YAML location is /codebuild/output/src960133886/src/github.com/cmilam87/runway/buildspec.yml 
[Container] 2019/08/13 21:12:21 No commands found for phase name: INSTALL 
[Container] 2019/08/13 21:12:21 Processing environment variables 
[Container] 2019/08/13 21:12:21 Moving to directory /codebuild/output/src960133886/src/github.com/cmilam87/runway 
[Container] 2019/08/13 21:12:21 Registering with agent 
[Container] 2019/08/13 21:12:21 Phases found in YAML: 2 
[Container] 2019/08/13 21:12:21  INSTALL: 0 commands 
[Container] 2019/08/13 21:12:21  BUILD: 3 commands 
[Container] 2019/08/13 21:12:21 Phase complete: DOWNLOAD_SOURCE State: SUCCEEDED 
[Container] 2019/08/13 21:12:21 Phase context status code:  Message:  
[Container] 2019/08/13 21:12:22 Entering phase INSTALL 
[Container] 2019/08/13 21:12:22 Running command echo "Installing Python version 3.7 ..." 
Installing Python version 3.7 ... 
 
[Container] 2019/08/13 21:12:22 Phase complete: INSTALL State: SUCCEEDED 
[Container] 2019/08/13 21:12:22 Phase context status code:  Message:  
[Container] 2019/08/13 21:12:22 Entering phase PRE_BUILD 
[Container] 2019/08/13 21:12:22 Phase complete: PRE_BUILD State: SUCCEEDED 
[Container] 2019/08/13 21:12:22 Phase context status code:  Message:  
[Container] 2019/08/13 21:12:22 Entering phase BUILD 
[Container] 2019/08/13 21:12:22 Running command cd $CODEBUILD_SRC_DIR/integration_tests 
 
[Container] 2019/08/13 21:12:22 Running command pipenv sync 
Creating a virtualenv for this project… 
Pipfile: /codebuild/output/src960133886/src/github.com/cmilam87/runway/integration_tests/Pipfile 
Using /usr/local/bin/python3 (3.7.3) to create virtualenv… 
Already using interpreter /usr/local/bin/python3 
Using base prefix '/usr/local' 
New python executable in /root/.local/share/virtualenvs/integration_tests-IsvfcGGK/bin/python3 
Also creating executable in /root/.local/share/virtualenvs/integration_tests-IsvfcGGK/bin/python 
Installing setuptools, pip, wheel... 
done. 
Running virtualenv with interpreter /usr/local/bin/python3 
 
Virtualenv location: /root/.local/share/virtualenvs/integration_tests-IsvfcGGK 
Installing dependencies from Pipfile.lock (9208d0)… 
To activate this project's virtualenv, run pipenv shell. 
Alternatively, run a command inside the virtualenv with pipenv run. 
All dependencies are now up-to-date! 
 
[Container] 2019/08/13 21:13:38 Running command pipenv run python runner.py 
INFO:testsuite:Loading tests from "/codebuild/output/src960133886/src/github.com/cmilam87/runway/integration_tests" with pattern: 
INFO:testsuite:Found test: "test_terraform.test_terraform". Attempting to import... 
INFO:testsuite:Executing test "Terraform"... 
INFO:testsuite:Executing "init" for "Terraform"... 
INFO:testsuite:Running "build" on tf_state.cfn ... 
[2019-08-13T21:13:39] Using interactive AWS provider mode. 
[2019-08-13T21:13:39] tf-state: submitted (creating new stack) 
[2019-08-13T21:14:39] tf-state: complete (creating new stack) 
INFO:testsuite:Loading tests from "/codebuild/output/src960133886/src/github.com/cmilam87/runway/integration_tests/test_terraform/tests" with pattern: 
INFO:testsuite:Found test: "test_terraform.tests.test_version". Attempting to import... 
INFO:testsuite:Found test: "test_terraform.tests.test_provider". Attempting to import... 
INFO:testsuite:Executing "run" for "Terraform"... 
INFO:testsuite:Executing test "VersionTest"... 
INFO:testsuite:Executing "init" for "VersionTest"... 
INFO:testsuite:Executing "run" for "VersionTest"... 
INFO:runway: 
INFO:runway:Environment "dev" was determined from the DEPLOY_ENVIRONMENT environment variable. 
INFO:runway:If this is not correct, update the value (or unset it to fall back to the name of the current git branch or parent directory). 
INFO:runway: 
INFO:runway:Found 1 deployment(s) 
INFO:runway: 
INFO:runway: 
INFO:runway:======= Processing deployment 'deployment_1' =========================== 
INFO:runway: 
INFO:runway:Attempting to deploy 'dev' to region(s): us-east-1 
INFO:runway: 
INFO:runway:======= Processing region us-east-1 =========================== 
INFO:runway: 
INFO:runway:---- Processing module 'terraform_test.tf' for 'dev' in us-east-1 -------------- 
INFO:runway:Module options: {'options': {'terraform_backend_config': {'region': 'us-east-1'}, 'terraform_backend_cfn_outputs': {'bucket': 'testsuite-tf-state::TerraformStateBucketName', 'dynamodb_table': 'testsuite-tf-state::TerraformLockTableName'}}, 'path': 'terraform_test.tf'} 
INFO:runway:Preparing to run terraform apply on terraform_test.tf... 
/root/.local/share/virtualenvs/integration_tests-IsvfcGGK/lib/python3.7/site-packages/botocore/vendored/requests/api.py:67: DeprecationWarning: You are using the get() function from 'botocore.vendored.requests'.  This is not a public API in botocore and will be removed in the future. Additionally, this version of requests is out of date.  We recommend you install the requests package, 'import requests' directly, and use the requests.get() function instead. 
  DeprecationWarning 
INFO:runway:Downloading and using Terraform version 0.11.13 ... 
INFO:runway:Downloaded Terraform 0.11.13 successfully 
INFO:runway:Running "terraform init"... 
INFO:runway:Using provided backend values "{'region': 'us-east-1', 'bucket': 'testsuite-tf-state', 'dynamodb_table': 'testsuite-tf-state-TerraformLockTable-OKYH2L6UZ30A'}" 
 
·[0m·[1mInitializing the backend...·[0m 
·[0m·[32m 
Successfully configured the backend "s3"! Terraform will automatically 
use this backend unless the backend configuration changes.·[0m 
 
·[0m·[1mInitializing provider plugins...·[0m 
- Checking for available provider plugins on https://releases.hashicorp.com... 
- Downloading plugin for provider "aws" (2.23.0)... 
 
·[0m·[1m·[32mTerraform has been successfully initialized!·[0m·[32m·[0m 
·[0m·[32m 
You may now begin working with Terraform. Try running "terraform plan" to see 
any changes that are required for your infrastructure. All Terraform commands 
should now work. 
 
If you ever set or change modules or backend configuration for Terraform, 
rerun this command to reinitialize your working directory. If you forget, other 
commands will detect it and remind you to do so if necessary.·[0m 
INFO:runway:Terraform workspace currently set to default; switching to dev... 
INFO:runway:Terraform workspace dev not found; creating it... 
·[0m·[32m·[1mCreated and switched to workspace "dev"!·[0m·[32m 
 
You're now on a new, empty workspace. Workspaces isolate their state, 
so if you run "terraform plan" Terraform will not see any existing state 
for this configuration.·[0m 
INFO:runway:Re-running terraform init after workspace change... 
INFO:runway:Using provided backend values "{'region': 'us-east-1', 'bucket': 'testsuite-tf-state', 'dynamodb_table': 'testsuite-tf-state-TerraformLockTable-OKYH2L6UZ30A'}" 
 
·[0m·[1mInitializing the backend...·[0m 
·[0m·[32m 
Successfully configured the backend "s3"! Terraform will automatically 
use this backend unless the backend configuration changes.·[0m 
 
·[0m·[1mInitializing provider plugins...·[0m 
 
·[0m·[1m·[32mTerraform has been successfully initialized!·[0m·[32m·[0m 
·[0m·[32m 
You may now begin working with Terraform. Try running "terraform plan" to see 
any changes that are required for your infrastructure. All Terraform commands 
should now work. 
 
If you ever set or change modules or backend configuration for Terraform, 
rerun this command to reinitialize your working directory. If you forget, other 
commands will detect it and remind you to do so if necessary.·[0m 
INFO:runway:Executing "terraform get" to update remote modules 
INFO:runway:Running Terraform apply on terraform_test.tf ("/root/.tfenv/versions/0.11.13/terraform apply -auto-approve=true -var-file=dev-us-east-1.tfvars") 
·[0m·[1maws_sqs_queue.terraform_queue: Creating...·[0m 
  arn:                               "" => "<computed>" 
  content_based_deduplication:       "" => "false" 
  delay_seconds:                     "" => "90" 
  fifo_queue:                        "" => "false" 
  kms_data_key_reuse_period_seconds: "" => "<computed>" 
  max_message_size:                  "" => "262144" 
  message_retention_seconds:         "" => "345600" 
  name:                              "" => "<computed>" 
  policy:                            "" => "<computed>" 
  receive_wait_time_seconds:         "" => "0" 
  visibility_timeout_seconds:        "" => "30"·[0m 
·[0m·[1maws_sqs_queue.terraform_queue: Creation complete after 0s (ID: https://sqs.us-east-1.amazonaws.com/923...7/terraform-20190813211452139300000001)·[0m·[0m 
·[0m·[1m·[32m 
Apply complete! Resources: 1 added, 0 changed, 0 destroyed.·[0m 
INFO:runway: 
INFO:runway:Environment "dev" was determined from the DEPLOY_ENVIRONMENT environment variable. 
INFO:runway:If this is not correct, update the value (or unset it to fall back to the name of the current git branch or parent directory). 
INFO:runway: 
INFO:runway:Found 1 deployment(s) 
INFO:runway: 
INFO:runway: 
INFO:runway:======= Processing deployment 'deployment_1' =========================== 
INFO:runway: 
INFO:runway:Attempting to deploy 'dev' to region(s): us-east-1 
INFO:runway: 
INFO:runway:======= Processing region us-east-1 =========================== 
INFO:runway: 
INFO:runway:---- Processing module 'terraform_test.tf' for 'dev' in us-east-1 -------------- 
INFO:runway:Module options: {'options': {'terraform_backend_config': {'region': 'us-east-1'}, 'terraform_backend_cfn_outputs': {'bucket': 'testsuite-tf-state::TerraformStateBucketName', 'dynamodb_table': 'testsuite-tf-state::TerraformLockTableName'}}, 'path': 'terraform_test.tf'} 
INFO:runway:Preparing to run terraform apply on terraform_test.tf... 
/root/.local/share/virtualenvs/integration_tests-IsvfcGGK/lib/python3.7/site-packages/botocore/vendored/requests/api.py:67: DeprecationWarning: You are using the get() function from 'botocore.vendored.requests'.  This is not a public API in botocore and will be removed in the future. Additionally, this version of requests is out of date.  We recommend you install the requests package, 'import requests' directly, and use the requests.get() function instead. 
  DeprecationWarning 
INFO:runway:Downloading and using Terraform version 0.12.6 ... 
INFO:runway:Downloaded Terraform 0.12.6 successfully 
INFO:runway:Running "terraform init"... 
INFO:runway:Using provided backend values "{'region': 'us-east-1', 'bucket': 'testsuite-tf-state', 'dynamodb_table': 'testsuite-tf-state-TerraformLockTable-OKYH2L6UZ30A'}" 
 
·[0m·[1mInitializing the backend...·[0m 
·[0m·[32m 
Successfully configured the backend "s3"! Terraform will automatically 
use this backend unless the backend configuration changes.·[0m 
 
·[0m·[1mInitializing provider plugins...·[0m 
 
·[0m·[1m·[32mTerraform has been successfully initialized!·[0m·[32m·[0m 
·[0m·[32m 
You may now begin working with Terraform. Try running "terraform plan" to see 
any changes that are required for your infrastructure. All Terraform commands 
should now work. 
 
If you ever set or change modules or backend configuration for Terraform, 
rerun this command to reinitialize your working directory. If you forget, other 
commands will detect it and remind you to do so if necessary.·[0m 
INFO:runway:Executing "terraform get" to update remote modules 
INFO:runway:Running Terraform apply on terraform_test.tf ("/root/.tfenv/versions/0.12.6/terraform apply -auto-approve=true -var-file=dev-us-east-1.tfvars") 
·[0m·[1maws_sqs_queue.terraform_queue: Refreshing state... [id=https://sqs.us-east-1.amazonaws.com/923778327047/terraform-20190813211452139300000001]·[0m 
·[0m·[1m·[32m 
Apply complete! Resources: 0 added, 0 changed, 0 destroyed.·[0m 
INFO:testsuite:Executing "teardown" for "VersionTest"... 
INFO:testsuite:Tearing down: test_terraform.tests.test_version 
INFO:runway: 
INFO:runway:Environment "dev" was determined from the DEPLOY_ENVIRONMENT environment variable. 
INFO:runway:If this is not correct, update the value (or unset it to fall back to the name of the current git branch or parent directory). 
INFO:runway: 
INFO:runway:WARNING! 
INFO:runway:Runway is running in DESTROY mode. 
INFO:runway:Found 1 deployment(s) 
INFO:runway: 
INFO:runway: 
INFO:runway:======= Processing deployment 'deployment_1' =========================== 
INFO:runway: 
INFO:runway:Attempting to deploy 'dev' to region(s): us-east-1 
INFO:runway: 
INFO:runway:======= Processing region us-east-1 =========================== 
INFO:runway: 
INFO:runway:---- Processing module 'terraform_test.tf' for 'dev' in us-east-1 -------------- 
INFO:runway:Module options: {'options': {'terraform_backend_config': {'region': 'us-east-1'}, 'terraform_backend_cfn_outputs': {'bucket': 'testsuite-tf-state::TerraformStateBucketName', 'dynamodb_table': 'testsuite-tf-state::TerraformLockTableName'}}, 'path': 'terraform_test.tf'} 
INFO:runway:Preparing to run terraform destroy on terraform_test.tf... 
INFO:runway:Terraform version 0.12.6 already installed; using it... 
INFO:runway:Running "terraform init"... 
INFO:runway:Using provided backend values "{'region': 'us-east-1', 'bucket': 'testsuite-tf-state', 'dynamodb_table': 'testsuite-tf-state-TerraformLockTable-OKYH2L6UZ30A'}" 
 
·[0m·[1mInitializing the backend...·[0m 
·[0m·[32m 
Successfully configured the backend "s3"! Terraform will automatically 
use this backend unless the backend configuration changes.·[0m 
 
·[0m·[1mInitializing provider plugins...·[0m 
 
·[0m·[1m·[32mTerraform has been successfully initialized!·[0m·[32m·[0m 
·[0m·[32m 
You may now begin working with Terraform. Try running "terraform plan" to see 
any changes that are required for your infrastructure. All Terraform commands 
should now work. 
 
If you ever set or change modules or backend configuration for Terraform, 
rerun this command to reinitialize your working directory. If you forget, other 
commands will detect it and remind you to do so if necessary.·[0m 
INFO:runway:Executing "terraform get" to update remote modules 
INFO:runway:Running Terraform destroy on terraform_test.tf ("/root/.tfenv/versions/0.12.6/terraform destroy -force -var-file=dev-us-east-1.tfvars") 
·[0m·[1maws_sqs_queue.terraform_queue: Refreshing state... [id=https://sqs.us-east-1.amazonaws.com/923778327047/terraform-20190813211452139300000001]·[0m 
·[0m·[1maws_sqs_queue.terraform_queue: Destroying... [id=https://sqs.us-east-1.amazonaws.com/923778327047/terraform-20190813211452139300000001]·[0m·[0m 
·[0m·[1maws_sqs_queue.terraform_queue: Destruction complete after 0s·[0m·[0m 
·[0m·[1m·[32m 
Destroy complete! Resources: 1 destroyed.·[0m 
INFO:testsuite:Executing test "ProviderTest"... 
INFO:testsuite:Executing "init" for "ProviderTest"... 
INFO:testsuite:Executing "run" for "ProviderTest"... 
INFO:runway: 
INFO:runway:Environment "dev" was determined from the DEPLOY_ENVIRONMENT environment variable. 
INFO:runway:If this is not correct, update the value (or unset it to fall back to the name of the current git branch or parent directory). 
INFO:runway: 
INFO:runway:Found 1 deployment(s) 
INFO:runway: 
INFO:runway: 
INFO:runway:======= Processing deployment 'deployment_1' =========================== 
INFO:runway: 
INFO:runway:Attempting to deploy 'dev' to region(s): us-east-1 
INFO:runway: 
INFO:runway:======= Processing region us-east-1 =========================== 
INFO:runway: 
INFO:runway:---- Processing module 'terraform_test.tf' for 'dev' in us-east-1 -------------- 
INFO:runway:Module options: {'options': {'terraform_backend_config': {'region': 'us-east-1'}, 'terraform_backend_cfn_outputs': {'bucket': 'testsuite-tf-state::TerraformStateBucketName', 'dynamodb_table': 'testsuite-tf-state::TerraformLockTableName'}}, 'path': 'terraform_test.tf'} 
INFO:runway:Preparing to run terraform apply on terraform_test.tf... 
INFO:runway:Terraform version 0.11.13 already installed; using it... 
INFO:runway:Running "terraform init"... 
INFO:runway:Using provided backend values "{'region': 'us-east-1', 'bucket': 'testsuite-tf-state', 'dynamodb_table': 'testsuite-tf-state-TerraformLockTable-OKYH2L6UZ30A'}" 
 
·[0m·[1mInitializing the backend...·[0m 
·[0m·[32m 
Successfully configured the backend "s3"! Terraform will automatically 
use this backend unless the backend configuration changes.·[0m 
 
·[0m·[1mInitializing provider plugins...·[0m 
- Checking for available provider plugins on https://releases.hashicorp.com... 
- Downloading plugin for provider "aws" (1.60.0)... 
 
·[0m·[1m·[32mTerraform has been successfully initialized!·[0m·[32m·[0m 
·[0m·[32m 
You may now begin working with Terraform. Try running "terraform plan" to see 
any changes that are required for your infrastructure. All Terraform commands 
should now work. 
 
If you ever set or change modules or backend configuration for Terraform, 
rerun this command to reinitialize your working directory. If you forget, other 
commands will detect it and remind you to do so if necessary.·[0m 
INFO:runway:Terraform workspace currently set to default; switching to dev... 
·[0m·[32mSwitched to workspace "dev".·[0m 
INFO:runway:Re-running terraform init after workspace change... 
INFO:runway:Using provided backend values "{'region': 'us-east-1', 'bucket': 'testsuite-tf-state', 'dynamodb_table': 'testsuite-tf-state-TerraformLockTable-OKYH2L6UZ30A'}" 
 
·[0m·[1mInitializing the backend...·[0m 
·[0m·[32m 
Successfully configured the backend "s3"! Terraform will automatically 
use this backend unless the backend configuration changes.·[0m 
·[31mError refreshing state: Terraform 0.11.13 does not support state version 4, please update.·[0m·[0m 
ERROR:testsuite:AssertionError: "test_terraform.tests.test_provider: Provider version 1 failed" 
INFO:testsuite:Executing "teardown" for "ProviderTest"... 
INFO:testsuite:Tearing down: test_terraform.tests.test_provider 
INFO:runway: 
INFO:runway:Environment "dev" was determined from the DEPLOY_ENVIRONMENT environment variable. 
INFO:runway:If this is not correct, update the value (or unset it to fall back to the name of the current git branch or parent directory). 
INFO:runway: 
INFO:runway:WARNING! 
INFO:runway:Runway is running in DESTROY mode. 
INFO:runway:Found 1 deployment(s) 
INFO:runway: 
INFO:runway: 
INFO:runway:======= Processing deployment 'deployment_1' =========================== 
INFO:runway: 
INFO:runway:Attempting to deploy 'dev' to region(s): us-east-1 
INFO:runway: 
INFO:runway:======= Processing region us-east-1 =========================== 
INFO:runway: 
INFO:runway:---- Processing module 'terraform_test.tf' for 'dev' in us-east-1 -------------- 
INFO:runway:Module options: {'options': {'terraform_backend_config': {'region': 'us-east-1'}, 'terraform_backend_cfn_outputs': {'bucket': 'testsuite-tf-state::TerraformStateBucketName', 'dynamodb_table': 'testsuite-tf-state::TerraformLockTableName'}}, 'path': 'terraform_test.tf'} 
INFO:runway:Preparing to run terraform destroy on terraform_test.tf... 
INFO:runway:Terraform version 0.11.13 already installed; using it... 
INFO:runway:Previous init failed; trashing .terraform directory... 
INFO:runway:Running "terraform init"... 
INFO:runway:Using provided backend values "{'region': 'us-east-1', 'bucket': 'testsuite-tf-state', 'dynamodb_table': 'testsuite-tf-state-TerraformLockTable-OKYH2L6UZ30A'}" 
 
·[0m·[1mInitializing the backend...·[0m 
·[0m·[32m 
Successfully configured the backend "s3"! Terraform will automatically 
use this backend unless the backend configuration changes.·[0m 
 
·[0m·[1mInitializing provider plugins...·[0m 
- Checking for available provider plugins on https://releases.hashicorp.com... 
- Downloading plugin for provider "aws" (1.60.0)... 
 
·[0m·[1m·[32mTerraform has been successfully initialized!·[0m·[32m·[0m 
·[0m·[32m 
You may now begin working with Terraform. Try running "terraform plan" to see 
any changes that are required for your infrastructure. All Terraform commands 
should now work. 
 
If you ever set or change modules or backend configuration for Terraform, 
rerun this command to reinitialize your working directory. If you forget, other 
commands will detect it and remind you to do so if necessary.·[0m 
INFO:runway:Terraform workspace currently set to default; switching to dev... 
·[0m·[32mSwitched to workspace "dev".·[0m 
INFO:runway:Re-running terraform init after workspace change... 
INFO:runway:Using provided backend values "{'region': 'us-east-1', 'bucket': 'testsuite-tf-state', 'dynamodb_table': 'testsuite-tf-state-TerraformLockTable-OKYH2L6UZ30A'}" 
 
·[0m·[1mInitializing the backend...·[0m 
·[0m·[32m 
Successfully configured the backend "s3"! Terraform will automatically 
use this backend unless the backend configuration changes.·[0m 
·[31mError refreshing state: Terraform 0.11.13 does not support state version 4, please update.·[0m·[0m 
INFO:testsuite:Executing "teardown" for "Terraform"... 
INFO:testsuite:Running "destroy" on tf_state.cfn ... 
[2019-08-13T21:15:21] Using interactive AWS provider mode. 
[2019-08-13T21:15:21] Executing pre_destroy hooks: runway.hooks.cleanup_s3.purge_bucket 
[2019-08-13T21:15:22] tf-state: submitted (submitted for destruction) 
[2019-08-13T21:16:22] tf-state: complete (stack destroyed) 
 
[Container] 2019/08/13 21:16:22 Phase complete: BUILD State: SUCCEEDED 
[Container] 2019/08/13 21:16:22 Phase context status code:  Message:  
[Container] 2019/08/13 21:16:22 Entering phase POST_BUILD 
[Container] 2019/08/13 21:16:22 Phase complete: POST_BUILD State: SUCCEEDED 
[Container] 2019/08/13 21:16:22 Phase context status code:  Message:  ```

File: .env [hunk 5]
```diff
@@ -1 +1 @@
-PYTHONPATH=src
+PYTHONPATH=src;integration_tests
```
Comment (troyready) [inline .env]: Will this work on Windows still if it's change to `PYTHONPATH=src:integration_tests` (semicolon -> colon)?

File: integration_tests/test_terraform/tf_state.cfn/templates/tf_state.yaml [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+      - !Ref 'BucketName'
+      - undefined
+  TableNameOmitted: !Or
+    - !Equals
+      - !Ref 'TableName'
+      - ''
+    - !Equals
+      - !Ref 'TableName'
+      - undefined
+Description: Create Terraform backend components - 1.19.0
```
Comment (troyready) [inline integration_tests/test_terraform/tf_state.cfn/templates/tf_state.yaml]: We can drop the "version number" from this description now

**Comment by troyready** [comment]
Oh, and I forgot -- we should remove the rest of the pyhcl stuff -- from setup.py and embedded folder

File: integration_tests/test_terraform/tf_state.cfn/templates/tf_state.yaml (hunk 6 above)
Comment (cmilam87) [reply integration_tests/test_terraform/tf_state.cfn/templates/tf_state.yaml]: Fixed.

File: .env (hunk 5 above)
Comment (cmilam87) [reply .env]: Yes, fixed.

**Comment by cmilam87** [comment]
> Oh, and I forgot -- we should remove the rest of the pyhcl stuff -- from setup.py and embedded folder

remove the install dependency and this folder? 
`src/runway/embedded/hcl` ?

**Comment by troyready** [comment]
> > Oh, and I forgot -- we should remove the rest of the pyhcl stuff -- from setup.py and embedded folder
> 
> remove the install dependency and this folder?
> `src/runway/embedded/hcl` ?

@cmilam87 yeah, exactly
